cipher/src/stream.rs

```diff
@@ -130,6 +140,16 @@ pub trait StreamCipher {
         self.try_apply_keystream(buf).unwrap();
     }
 
+    /// Write keystream to `buf`.
+    ///
+    /// # Panics
+    /// If end of the keystream will be reached with the given data length,
+    /// method will panic without modifying the provided `data`.
+    #[inline]
+    fn write_keystream(&mut self, buf: &mut [u8]) {
```

Comment:
Any strong reason to prefer `&mut [u8]` over `&mut [MaybeUninit<u8>]`? That is exactly the type that `Vec::spare_capacity_mut()` returns and I'd rather not rely on compiler guessing and removing extra zeroing.

In fact the method could me something like this:
```rust
fn write_keystream(&mut self, buf: &mut [MaybeUninit<u8>]) -> &mut [u8] {
```

Where return type is `buf` with bytes filled in.